src/Serializers/Orleans.Serialization.Protobuf/ProtobufCodec.cs

```diff
@@ -95,16 +97,34 @@ bool IGeneralizedCopier.IsSupportedType(Type type)
     /// <inheritdoc/>
     bool? ITypeFilter.IsTypeAllowed(Type type)
     {
-        if (!typeof(IMessage).IsAssignableFrom(type))
+        if (!MessageType.IsAssignableFrom(type))
         {
             return null;
         }
 
+        if (type == MessageType)
+        {
+            // While IMessage is the basis of all supported types, it isn't directly supported
+            return null;
+        }
+
         return ((IGeneralizedCodec)this).IsSupportedType(type) || ((IGeneralizedCopier)this).IsSupportedType(type);
-    } 
+    }
 
     private static bool IsProtobufMessage(Type type)
     {
+        if (type == MessageType)
+        {
+            // Not a concrete implementation, so not directly serializable
+            return false;
+        }
+
+        if (type == MessageGenericType)
+        {
+            // Not a concrete implementation, but the generic type does give the concrete type
+            type = type.GenericTypeArguments[0];
+        }
+
         if (!MessageParsers.ContainsKey(type.TypeHandle))
         {
             if (Activator.CreateInstance(type) is not IMessage protobufMessageInstance)
```

Comment:
It seems that this check could be performed via `type.IsAssignableTo(typeof(IMessage))` without invoking the ctor

Reply:
This method is really making sure that a message parser can be obtained (and caching that value) as opposed to just verifying the interface. The only other way to obtain the parser is via reflection.